morf-core/src/main/java/org/alfasoftware/morf/jdbc/SqlScriptExecutor.java

```diff
@@ -66,13 +69,37 @@ public class SqlScriptExecutor {
     super();
     this.dataSource = dataSource;
     this.sqlDialect = sqlDialect;
+    this.fetchSizeForBulkSelects = sqlDialect.fetchSizeForBulkSelects();
+    this.fetchSizeForBulkSelectsAllowingConnectionUseDuringStreaming = sqlDialect.fetchSizeForBulkSelectsAllowingConnectionUseDuringStreaming();
     if (visitor == null) {
       this.visitor = new NullVisitor();
     } else {
       this.visitor = visitor;
     }
   }
 
+  /**
+   * Create an SQL executor with the given visitor, who will
+   * be notified about events.
+   *
+   * @param visitor Visitor to notify about execution.
+   * @param dataSource DataSource to use.
+   * @param connectionResources The connection resources to use.
+   */
+  SqlScriptExecutor(SqlScriptVisitor visitor, DataSource dataSource, SqlDialect sqlDialect, ConnectionResources connectionResources) {
+    super();
+    this.dataSource = dataSource;
+    this.sqlDialect = sqlDialect;
+    this.fetchSizeForBulkSelects = connectionResources.getFetchSizeForBulkSelects() != null
+        ? connectionResources.getFetchSizeForBulkSelects() : sqlDialect.fetchSizeForBulkSelects();
+    this.fetchSizeForBulkSelectsAllowingConnectionUseDuringStreaming = connectionResources.getFetchSizeForBulkSelectsAllowingConnectionUseDuringStreaming() != null
+        ? connectionResources.getFetchSizeForBulkSelectsAllowingConnectionUseDuringStreaming() : sqlDialect.fetchSizeForBulkSelectsAllowingConnectionUseDuringStreaming();
+    if (visitor == null) {
+      this.visitor = new NullVisitor();
+    } else {
+      this.visitor = visitor;
+    }
+  }
```

Comment:
We are duplicating the null visitor code here. While rather minor thing right now, it feels like a step in the wrong direction.

We could either extract a common private contructor to be called from both of these.
Or we could extract the `if (visitor == null)` logic into a private static method to be called from both of these.
The latter is probably better for the future?